packages/common/src/core/schemas/shared/subschemas.ts

```diff
@@ -100,3 +104,78 @@ export const ENTITY_TIMELINE_SCHEMA = {
     },
   },
 };
+
+/**
+ * Returns the UI schema elements necessary to properly render
+ * the thumbnail editing control for an item-based entity.
+ *
+ * @param i18nScope i18n scope for the entity translations
+ * @param entity The entity to build the UI schema for
+ * @returns an array of UI schema elements for thumbnail editing
+ */
+export function getThumbnailUiSchemaElements(
+  i18nScope: string,
+  entity: IHubItemEntity
+): IUiSchemaElement[] {
+  return [
+    // HIDDEN, READ-ONLY CONTROLS FOR VALIDATION RULES
```

Comment:
For posterity, we ended up, not taking this approach as it might incentivize developers to use this custom function rather than expressing their conditions with JSON schema.